assertion_test.go

```diff
@@ -195,7 +195,205 @@ func TestAssertion_HandlerPanics(t *testing.T) {
+				Actual: nil,
+			},
+			traits: fieldTraits{
+				Actual: fieldRequired,
+			},
+		},
+		{
+			testName:          "denied Actual",
+			errorContainsText: "Actual",
+			failure: AssertionFailure{
+				Actual: &AssertionValue{},
+			},
+			traits: fieldTraits{
+				Actual: fieldDenied,
+			},
+		},
+		{
+			testName:          "required Expected",
+			errorContainsText: "Expected",
+			failure: AssertionFailure{
+				Expected: nil,
+			},
+			traits: fieldTraits{
+				Expected: fieldRequired,
+			},
+		},
+		{
+			testName:          "denied Expected",
+			errorContainsText: "Expected",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{},
+			},
+			traits: fieldTraits{
+				Expected: fieldDenied,
+			},
+		},
+		{
+			testName:          "required on Expected",
+			errorContainsText: "",
+			failure: AssertionFailure{
+				Expected: nil,
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "AssertionRange should not be pointer",
+			errorContainsText: "AssertionRange",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: &AssertionRange{},
+				},
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "Min should be non-nil",
+			errorContainsText: "AssertionRange",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionRange{
+						Max: 1,
+					},
+				},
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "Max should be non-nil",
+			errorContainsText: "AssertionRange",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionRange{
+						Min: 1,
+					},
+				},
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "AssertionList should not be pointer",
+			errorContainsText: "AssertionValue",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: &AssertionList{},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+		{
+			testName:          "AssertionList should be pointer",
+			errorContainsText: "AssertionList",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: &AssertionList{},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+		{
+			testName:          "AssertionList should be not nil",
+			errorContainsText: "AssertionList",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionList{},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+		{
+			testName:          "Expected can't be slice of single element",
+			errorContainsText: "",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionList{[]int{1}},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+		{
+			testName:          "required List",
+			errorContainsText: "",
```

Comment:
Let's rename this to `required AssertionList` for consistency.

Also, errorContainsText can be set to "Expected".